### ISSUE TITLE
Scheduler: Stop swallowing invalid healthz and metrics address for scheduler config

### ISSUE THREAD

**Issue by gavinfish** [kubernetes/kubernetes#97048]
**What would you like to be added**:

When scheduler config provides invalid healthz and metrics address, instead of swallowing the invalid config by setting a default value for it, it is better to just throw the error. And in validation, we have already had validation for this cases. So what I want to achieve here is that skipping to set default value for healthz and metrics address in v1beta2 when they are totally invalid.

Invalid address here means that the address can not be parsed as valid host and port, or either of them is valid. Will not change current default setting flow:

```
	// 1. If the value is nil, default to 0.0.0.0 and default scheduler port
	// 2. If there is a value set, attempt to split it. If it's just a port (ie, ":1234"), default to 0.0.0.0 with that port
	// 3. If splitting the value fails, check if the value is even a valid IP. If so, use that with the default port.
```

/cc @damemi  @alculquicondor 
/assign

**Comment by alculquicondor** [comment]
Please mention v1beta2 in the description.

**Comment by gavinfish** [comment]
> /assign
> 
> self-assigning this if that's alright with you, @gavinfish, because I'm kinda working on this in #99597

sure, please go ahead.

**Comment by alculquicondor** [comment]
Fixed in #99597

/close
